src/de.rs

```diff
@@ -1451,13 +1454,27 @@ impl<'a> Deserializer<'a> {
                         expected: "a value",
                         found: token.1.describe(),
                     },
-                ))
+                ));
             }
             None => return Err(self.eof()),
         };
         Ok(value)
     }
 
+    fn parse_keylike(&mut self, at: usize, span: Span, key: &'a str) -> Result<Value<'a>, Error> {
+        let lowercase = key.to_ascii_lowercase();
+        let lowercase = lowercase.as_str();
+        if lowercase == "inf" || lowercase == "nan" {
```

Comment:
Just a quick question:  What is the reason to use lowercase here?  It's probably not too expensive, but it seems like an unnecessary expense for an obscure edge case.

Reply:
I though all of the spellings are valid and wanted to check those.

Checked the spec:
> Special float values can also be expressed. They are always lowercase.

https://toml.io/en/v1.0.0-rc.3#user-content-float
I was wrong and I must not convert strings to lowercase. Will fix.

Reply:
Fixed.